contrib/win32/win32compat/signal.c

```diff
@@ -260,59 +260,43 @@ sw_process_pending_signals()
 int
 wait_for_any_event(HANDLE* events, int num_events, DWORD milli_seconds)
 {
-	HANDLE all_events[MAXIMUM_WAIT_OBJECTS];
-	DWORD num_all_events;
+	HANDLE all_events[MAX_CHILDREN];
```

Comment:
Your call.  Performance of stack memory operations are typically constant (i.e. independent of size) since you just end up moving a single pointer higher/lower in the stack as you "allocate" and "free"; how much you move the pointer doesn't really matter unless you're worried about recursion / blowing your stack which is not a concern in this case.   I think the risk of marking it static and making function no longer thread-safe and would outweigh any potential gain.